src/pages/phy-count-serialSum/index.js

```diff
@@ -0,0 +1,401 @@
+    },
+    maxAccess,
+    enableReinitialize: true,
+    validateOnChange: true,
+    validationSchema: yup.object({
+      stockCountId: yup.string().required(),
+      siteId: yup.string().required()
+    })
+  })
+
+  const handleSearchChange = event => {
+    const { value } = event.target
+    formik.setFieldValue('search', value)
+  }
+
+  async function fetchGridData() {
+    if (!formik.values.stockCountId || !formik.values.siteId) return
+
+    const res = await getRequest({
+      extension: SCRepository.StockCountSerialSum.qry,
+      parameters: `_stockCountId=${formik.values.stockCountId}&_siteId=${formik.values.siteId}`
+    })
+
+    let sumPcs = 0
+    let sumWeight = 0
+    let sumSystemPcs = 0
+    let sumVariancePcs = 0
+    let sumVarianceWght = 0
+
+    const updatedList = res.list.map(item => {
+      sumPcs += item.countedPcs || 0
+      sumWeight += item.weight || 0
+      sumSystemPcs += item.systemPcs || 0
+      sumVariancePcs += item.variancePcs || 0
+      sumVarianceWght = sumWeight * sumVariancePcs
+
+      return {
+        ...item,
+        netWeight: item.weight * item.countedPcs,
+        varianceWght: item.weight * item.variancePcs
+      }
+    })
+
+    formik.setFieldValue('totalCountedPcs', sumPcs)
+    formik.setFieldValue('totalWeight', sumWeight)
+    formik.setFieldValue('totalSystemPcs', sumSystemPcs)
+    formik.setFieldValue('totalVariancePcs', sumVariancePcs)
+    formik.setFieldValue('totalVarianceWeight', sumVarianceWght)
+
+    setData({ list: updatedList })
+    handleClick(updatedList)
+  }
+
+  const fillSiteStore = stockCountId => {
+    setSiteStore([])
+    var parameters = `_stockCountId=${stockCountId}`
+    getRequest({
+      extension: SCRepository.Sites.qry,
+      parameters: parameters
+    }).then(res => {
+      setSiteStore(res.list)
+    })
+  }
+
+  const columns = [
+    {
+      field: 'srlNo',
+      headerName: _labels.srlNo,
+      flex: 1
+    },
+    {
+      field: 'sku',
+      headerName: _labels.sku,
+      flex: 1
+    },
+    {
+      field: 'itemName',
+      headerName: _labels.name,
+      flex: 1
+    },
+    {
+      field: 'countedPcs',
+      headerName: _labels.countedPcs,
+      flex: 1,
+      type: 'number'
+    },
+    {
+      field: 'systemPcs',
+      headerName: _labels.systemPcs,
+      flex: 1,
+      type: 'number'
+    },
+    {
+      field: 'variancePcs',
+      headerName: _labels.variancePcs,
+      flex: 1,
+      type: 'number'
+    },
+    {
+      field: 'weight',
+      headerName: _labels.weight,
+      flex: 1,
+      type: 'number'
+    },
+
+    {
+      field: 'varianceWght',
+      headerName: _labels.varW,
+      flex: 1,
+      type: 'number'
+    },
+    {
+      field: 'netWeight',
+      headerName: _labels.netWeight,
+      flex: 1,
+      type: 'number'
+    }
+  ]
+
+  useEffect(() => {
+    ;(async function () {
+      await fetchGridData()
+    })()
+  }, [formik.values.stockCountId, formik.values.siteId])
+
```

Comment:
check if removing the formik.values.stockCountId will affect the behaviour 
because we need to fill site and anyway it can't be filled without stock count 
so i guess no need to fetch data after choosing stock count
specially that after selections the fields will be readonly

Reply:
it cant be removing when there is no stoccount id the siteId is read only , and when we fill siteId stockcount will be read only and site id read only also , to change we have to press clear like asp